pkg/operator/mon/mon.go

```diff
@@ -85,8 +89,47 @@ func (c *Cluster) Start(clientset *kubernetes.Clientset) (*mon.ClusterInfo, erro
 
 // Retrieve the ceph cluster info if it already exists.
 // If a new cluster create new keys.
-func (c *Cluster) initClusterInfo() (*mon.ClusterInfo, error) {
-	return mon.CreateClusterInfo(c.factory, "")
+func (c *Cluster) initClusterInfo(clientset *kubernetes.Clientset) (*mon.ClusterInfo, error) {
+	secrets, err := clientset.Secrets(c.Namespace).Get(monSecretsName)
+	if err != nil {
+		if !k8sutil.IsKubernetesResourceNotFoundError(err) {
+			return nil, fmt.Errorf("failed to get mon secrets. %+v", err)
+		}
+
+		return c.createMonSecretsAndSave(clientset)
+	}
+
+	info := &mon.ClusterInfo{
+		Name:          string(secrets.Data[clusterSecretName]),
+		FSID:          string(secrets.Data[fsidSecretName]),
+		MonitorSecret: string(secrets.Data[monSecretName]),
+		AdminSecret:   string(secrets.Data[adminSecretName]),
+		Monitors:      map[string]*mon.CephMonitorConfig{},
+	}
+	logger.Infof("found existing monitor secrets for cluster %s with fsid %s. %+v", info.Name, info.FSID, info)
+	return info, nil
+}
+
+func (c *Cluster) createMonSecretsAndSave(clientset *kubernetes.Clientset) (*mon.ClusterInfo, error) {
+	logger.Infof("creating mon secrets for a new cluster")
+	info, err := mon.CreateClusterInfo(c.factory, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create mon secrets. %+v", err)
+	}
+
+	// store the secrets
+	secrets := map[string]string{
+		clusterSecretName: info.Name,
+		fsidSecretName:    info.FSID,
+		monSecretName:     info.MonitorSecret,
+		adminSecretName:   info.AdminSecret,
+	}
+	_, err = clientset.Secrets(c.Namespace).Create(&v1.Secret{ObjectMeta: v1.ObjectMeta{Name: monSecretsName}, StringData: secrets})
```

Comment:
bah, i intended to stop writing them to the log when i was done debugging. will remove